auth/auth/auth.py

```diff
@@ -68,39 +68,15 @@
 routes = web.RouteTableDef()
 
 
-def authenticated_users_only(fun: AuthenticatedAIOHTTPHandler) -> AIOHTTPHandler:
-    @wraps(fun)
-    async def wrapped(request: web.Request) -> web.StreamResponse:
+class LocalAuthenticator(Authenticator):
+    async def _fetch_userdata(self, request: web.Request) -> Optional[UserData]:
         session_id = await get_session_id(request)
```

Comment:
During review of this I noticed that `get_session_id` and `_get_session_id` are pretty similar but have a different preference order for cookie vs header. Can we make these methods more similar? I understand that services other than auth won't look at X-Hail-Internal-Authorization, but it seems like the auth-only get_session_id should delegate to a common get_session_id if we're not using X-Hail-Internal-Authorization, right?

Reply:
Good point, I've moved `auth` to check for a `X-Hail-Internal-Authorization` header when appropriate and fall back on the `get_session_id` in gear.